src/daemon/rpmostree-sysroot-upgrader.cxx

```diff
@@ -1439,9 +1440,17 @@ rpmostree_sysroot_upgrader_deploy (RpmOstreeSysrootUpgrader *self,
   if (g_hash_table_size (rpmostree_origin_get_initramfs_etc_files (self->origin)) > 0)
     {
       glnx_fd_close int fd = -1;
-      if (!ror_initramfs_overlay_generate (rpmostree_origin_get_initramfs_etc_files (self->origin),
-                                           &fd, cancellable, error))
-        return glnx_prefix_error (error, "Generating initramfs overlay");
+      auto etc_files_gset = rpmostree_origin_get_initramfs_etc_files (self->origin);
+      rust::Vec<rust::String> etc_files;
```

Comment:
Do you know whether `cxx` supports C++ `std::set`, so that we could hopefully directly pass something like a  `BTreeSet` to Rust?

Reply:
No; see https://cxx.rs/bindings.html for the current list which is basically array/vector types.
But looking at https://github.com/dtolnay/cxx/pulls?q=is%3Apr+is%3Aclosed things are improving *fast* and it's on the TODO.

In this case though note that we aren't changing the ordering logic here - the origin API still returns it as a hash table.  And of course we're making progress on oxidizing the origin code...which could lead to us not even passing it across the C++ side at all, maintaining it as a rust `BTreeSet` from the start.